### PR TITLE
[test]Add the string testcase of index and query

### PR DESCRIPTION
**Pull request by jingkl**
Signed-off-by: jingkl <jingjing.jia@zilliz.com>
issue : #16833

### CODE REVIEW

File: tests/python_client/testcases/test_index.py [hunk 1]
```diff
@@ -1310,3 +1315,188 @@ def test_create_index_callback(self, connect, collection, get_simple_index):
...
+                2.create an index with string field after collection load
+        expected: create index successfully
+        """
+        c_name = cf.gen_unique_str(prefix)
+        collection_w = self.init_collection_wrap(name=c_name)
+        data = cf.gen_default_list_data(ct.default_nb)
+        collection_w.insert(data=data)
+        index, _ = self.index_wrap.init_index(collection_w.collection, default_string_field_name, default_string_index_params)
+        cf.assert_equal_index(index, collection_w.collection.indexes[0])
+        collection_w.load()
```
Comment (binbinlv) [inline tests/python_client/testcases/test_index.py:1355]: Here the implement of this test case is not consistent with what you described in test comment and test case name.

File: tests/python_client/testcases/test_index.py [hunk 2]
```diff
@@ -1310,3 +1315,188 @@ def test_create_index_callback(self, connect, collection, get_simple_index):
...
+        method: 1.create collection and insert data 
+                2.only create an index with string field is not primary
+        expected: create index successfully
+        """
+        c_name = cf.gen_unique_str(prefix)
+        collection_w = self.init_collection_wrap(name=c_name)
+        data = cf.gen_default_list_data()
+        collection_w.insert(data=data)
+        index, _ = self.index_wrap.init_index(collection_w.collection, default_string_field_name, default_string_index_params)
+        cf.assert_equal_index(index, collection_w.collection.indexes[0])
```
Comment (binbinlv) [inline tests/python_client/testcases/test_index.py:1339]: I think "collection_w.collection.indexes" should be replaced by "collection_w.indexes". And all of these code should be modified.

File: tests/python_client/testcases/test_query.py [hunk 3]
```diff
@@ -1190,3 +1195,88 @@ def test_query_growing_segment_data(self):
...
+        """
+        target: test query data with string field is not primary
+        method: create collection and insert data
+                collection.load()
+                query with string expr in string field is not primary
+        expected: query successfully
+        """
+     
+        collection_w, vectors = self.init_collection_general(prefix, insert_data=True)[0:2]
+        collection_w.load()
```
Comment (binbinlv) [inline tests/python_client/testcases/test_query.py:1217]: This load could be deleted.

File: tests/python_client/testcases/test_query.py [hunk 4]
```diff
@@ -1190,3 +1195,88 @@ def test_query_growing_segment_data(self):
...
+    def test_query_string_with_mix_expr(self):
+        """
+        target: test query data
+        method: create collection and insert data
+                collection.load()
+                query with mix expr in string field and int field
+        expected: query successfully
+        """
+        collection_w, vectors = self.init_collection_general(prefix, insert_data=True, primary_field=ct.default_string_field_name)[0:2]
+        collection_w.load()
```
Comment (binbinlv) [inline tests/python_client/testcases/test_query.py:1246]: This could be deleted.

File: tests/python_client/testcases/test_query.py [hunk 5]
```diff
@@ -1190,3 +1195,88 @@ def test_query_growing_segment_data(self):
...
+    def test_query_with_invalid_string_expr(self, expression):
+        """
+        target: test query data
+        method: create collection and insert data
+                collection.load()
+                query with invalid expr
+        expected: Raise exception
+        """
+        collection_w = self.init_collection_general(prefix, insert_data=True)[0]
+        collection_w.load()
```
Comment (binbinlv) [inline tests/python_client/testcases/test_query.py:1264]: This could be deleted.

File: tests/python_client/testcases/test_query.py [hunk 6]
```diff
@@ -1190,3 +1195,88 @@ def test_query_growing_segment_data(self):
...
+       
+
+    @pytest.mark.tags(CaseLabel.L1)
+    def test_query_string_expr_with_binary(self):
+        """
+        target: test query string expr with binary
+        method: query string expr with binary 
+        expected: verify query successfully
+        """
+        collection_w, vectors= self.init_collection_general(prefix, insert_data=True, is_binary=True)[0:2]
```
Comment (binbinlv) [inline tests/python_client/testcases/test_query.py:1276]: Here should add "is_index=True"

File: tests/python_client/testcases/test_query.py [hunk 7]
```diff
@@ -1190,3 +1195,88 @@ def test_query_growing_segment_data(self):
...
+        collection_w.create_index(ct.default_binary_vec_field_name, binary_index_params)
```
Comment (binbinlv) [inline tests/python_client/testcases/test_query.py:1277]: After create_index, load() should be added if you want index you created is valid before query.

File: tests/python_client/testcases/test_index.py (hunk 1 above)
Comment (jingkl) [reply tests/python_client/testcases/test_index.py:1355]: I think binary is already indexed underneath, so this piece can be done without is index=true

**Pull request merged**